### DISCUSSION TITLE
Is there a way to listen for midi device connect events?

### DISCUSSION THREAD

**Discussion by jdlee022** [djipco/webmidi#244] [Q&A] [answered]
I'm looking for a way to handle an event when a new midi device (new input) is added). Is it somewhere in the docs? If so, im having trouble finding it.

**Comment by djipco** [comment, marked as answer]
The `WebMidi` object dispatches several [events](https://webmidijs.org/api/classes/WebMidi#events). Two of those are [`connected`](https://webmidijs.org/api/classes/WebMidi#event-connected) and [`disconnected`](https://webmidijs.org/api/classes/WebMidi#event-disconnected). They are triggered when a device is being connected or disconnected.
